components/service-operator/apis/database/v1beta1/postgres_types.go

```diff
@@ -279,6 +280,52 @@ func (p *Postgres) GetServiceEntrySpecs(outputs cloudformation.Outputs) ([]map[s
 	return specs, nil
 }
 
+// GetStackPolicy implements cloudformation.Stack to return a serialised form of the stack policy, or nil if one is
+// not needed
+func (p *Postgres) GetStackPolicy() aws.StackPolicyDocument {
+	statements := []aws.StatementEntry{
+		{
+			Effect:    "Deny",
+			Action:    []string{"Update:Replace", "Update:Delete"},
+			Principal: "*",
+			Resource:  "LogicalResourceId/RDSCluster",
+		},
+		{
+			Effect:    "Allow",
+			Action:    []string{"Update:Modify"},
+			Principal: "*",
+			Resource:  "LogicalResourceId/RDSCluster",
+		},
+	}
+
+	instanceCount := p.Spec.AWS.InstanceCount
+	if instanceCount < 1 {
+		instanceCount = DefaultInstanceCount
+	}
+
+	for i := 0; i < instanceCount; i++ {
+		statements = append(statements, aws.StatementEntry{
+			Effect:    "Deny",
+			Action:    []string{"Update:Replace", "Update:Delete"},
+			Principal: "*",
+			Resource:  fmt.Sprintf("LogicalResourceId/%s%d", PostgresResourceInstance , i),
+		})
+
+		statements = append(statements, aws.StatementEntry{
+			Effect:    "Allow",
```

Comment:
I'm a bit worried about this Allow with Principal *. Is the default to allow stuff where not otherwise denied? If so then this is unnecessary. If not then we should probably allow for all resources. As is this feels like it might allow any principal to modify our RDS Instances?

Reply:
According to the docs, that value has to be '*'.

Reply:
Spoke to Daniel, we think the answer is that the default is to allow stuff where not otherwise denied, these Allow statements are here for clarity's sake. Principal can only be * and that's not going to have any cross-account implications.